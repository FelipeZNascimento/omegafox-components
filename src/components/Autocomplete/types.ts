export type TDropdownItem = {
  id: number;
  name: string;
  abbreviationEn: string;
};

export interface IAutocompleteProps {
  dropdownList: TDropdownItem[];
  onSelect: (item: TDropdownItem) => void;
}
