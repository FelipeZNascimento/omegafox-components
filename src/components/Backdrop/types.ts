export interface IBackdropProps {
  align?: 'left' | 'center' | 'right';
  children: JSX.Element;
  isOpen: boolean;
}
