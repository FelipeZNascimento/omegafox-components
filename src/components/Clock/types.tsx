import {
  FOOTBALL_MATCH_STATUS,
  AMERICAN_FOOTBALL_MATCH_STATUS
} from './constants';
import { TBetValues } from '../types';

type TClock = {
  time: number;
};

type FootballKeys = keyof typeof FOOTBALL_MATCH_STATUS;
export type TClockFootball = TClock & {
  status: typeof FOOTBALL_MATCH_STATUS[FootballKeys];
};

type AmericanFootballKeys = keyof typeof AMERICAN_FOOTBALL_MATCH_STATUS;
export type TClockAmericanFootball = TClock & {
  status: typeof AMERICAN_FOOTBALL_MATCH_STATUS[AmericanFootballKeys];
};

export interface IClockProps {
  betValue: TBetValues | null;
  clock: TClockAmericanFootball | TClockFootball;
  isMatchStarted?: boolean;
  isExpandable: boolean;
  isExpanded: boolean;
  isMatchEditable: boolean;
  timestamp: number;
}

export interface IClockMobileProps {
  clock: TClockAmericanFootball | TClockFootball;
  timestamp: number;
}
