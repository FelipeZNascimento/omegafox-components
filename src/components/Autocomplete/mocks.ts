import { TDropdownItem } from './types';

export const dropdownList: TDropdownItem[] = [
  {
    id: 0,
    name: 'Lionel Messi',
    abbreviationEn: 'ARG'
  },
  {
    id: 1,
    name: 'Cristiano Ronaldo',
    abbreviationEn: 'POR'
  },
  {
    id: 2,
    name: 'Mbappé',
    abbreviationEn: 'FRA'
  },
  {
    id: 3,
    name: 'Caio Jr.',
    abbreviationEn: 'BRA'
  },
  {
    id: 4,
    name: 'Darwin Nuñez',
    abbreviationEn: 'URU'
  }
];
