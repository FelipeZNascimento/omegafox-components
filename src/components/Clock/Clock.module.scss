@use "../../styling/colors.scss" as colors;
@use "../../styling/colors_fifa.scss" as colors_fifa;
@use "../../styling/color_utils.scss" as color_utils;

@import "../../styling/alignments.scss";
@import "../../styling/font-sizes.scss";
@import "../../styling/margins.scss";
@import "../../styling/paddings.scss";

.dateContainer {
  @include padded(horizontal, s2);
  @include marged(right, s3);

  &Desktop {
    @extend .dateContainer;

    height: $match-container-height;
    border-radius: $border-radius;
    border-bottom-left-radius: 0;
    position: relative;
    width: $clock-width;
  }

  &Mobile {
    @extend .dateContainer;

    border-radius: $border-radius;
    left: 10px;
    position: absolute;
    top: -16px;
    width: 100px;
  }

  .info {
    &Desktop {
      @include flex-aligned(center, center);
      height: 36px; //Without grey area
    }

    &Mobile {
      @include flex-aligned(space-evenly, flex-start);
      height: 36px;
    }

    &Date {
      @include font-sized(m);

      color: colors.$grey5;
      font-weight: bold;
    }

    &Clock {
      @include font-sized(m);

      &Desktop {
        @include font-sized(l);
        flex-direction: column;
      }

      &Mobile {
        @include font-sized(s);
        flex-direction: row;
      }

      &Status {
        @include font-sized(s);
      }

      &Animated {
        color: colors_fifa.$qatar-red;
        animation: pulse 2s ease-in-out infinite;
      }
    }
  }
}

.toggle {
  @include font-sized(l2);
  @include flex-aligned(center, center);

  border-bottom-right-radius: $border-radius;
  bottom: 0px;
  color: colors.$white;
  height: 24px;
  left: 0px;
  position: absolute;
  transition: filter $transition-time;
  width: 100%;

  &Neutral {
    background-color: rgba(colors.$black, 0.5);
  }

  &Green {
    background-color: rgba(colors.$mint, 0.6);
  }

  &Blue {
    background-color: rgba(colors_fifa.$qatar-blue, 0.6);
  }

  &LightBlue {
    background-color: rgba(colors_fifa.$qatar-lightblue, 0.6);
  }

  &Yellow {
    background-color: rgba(colors.$gold, 0.6);
  }

  &Red {
    background-color: rgba(colors_fifa.$qatar-red, 0.6);
  }
}

@keyframes pulse {
  0% {
    filter: brightness(80%);
  }

  50% {
    filter: brightness(150%);
  }

  100% {
    filter: brightness(80%);
  }
}
