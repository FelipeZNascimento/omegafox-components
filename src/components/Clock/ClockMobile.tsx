import { DateTime } from 'luxon';
import classNames from 'classnames';

import { IClockMobileProps } from './types';

import {
  LABELS,
  MATCH_STATUS,
  FOOTBALL_MATCH_STATUS,
  STOPPED_GAME
} from './constants';
import styles from './Clock.module.scss';
import matchLayer from '../../img/match_layer.png';

export const ClockMobile = ({ clock, timestamp }: IClockMobileProps) => {
  const formattedDate = DateTime.fromSeconds(timestamp)
    .setLocale('pt-Br')
    .toFormat("HH'h'mm");

  const dateClass = classNames(styles.infoMobile, {
    [styles.infoDate]: clock.status === MATCH_STATUS.NOT_STARTED,
    [styles.infoClockMobile]: clock.status !== MATCH_STATUS.NOT_STARTED
  });

  const clockClass = classNames({
    [styles.infoClockAnimated]: !STOPPED_GAME.includes(clock.status)
  });

  const renderDate = () => {
    if (clock.status === MATCH_STATUS.NOT_STARTED) {
      return formattedDate;
    } else {
      if (STOPPED_GAME.includes(clock.status)) {
        return LABELS[clock.status];
      } else {
        return (
          <>
            <span className={styles.infoClockStatus}>
              {clock.status !== FOOTBALL_MATCH_STATUS.PENALTIES &&
                LABELS[clock.status]}
            </span>
            <span className={clockClass}>
              {clock.status !== FOOTBALL_MATCH_STATUS.PENALTIES &&
                `${clock.time}'`}
              {clock.status === FOOTBALL_MATCH_STATUS.PENALTIES &&
                LABELS[clock.status]}
            </span>
          </>
        );
      }
    }
  };

  return (
    <div
      className={styles.dateContainerMobile}
      style={{
        background: `url(${matchLayer}) #eceff1`
      }}
    >
      <div className={dateClass}>{renderDate()}</div>
    </div>
  );
};
