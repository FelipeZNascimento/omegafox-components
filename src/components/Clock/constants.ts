export const MATCH_STATUS = {
  NOT_STARTED: 0,
  FINAL: 1,
  DELAYED: 2,
  CANCELLED: 3,
  OTHER: 4
};

export const FOOTBALL_MATCH_STATUS = {
  ...MATCH_STATUS,
  FIRST_HALF: 20,
  HALFTIME: 21,
  SECOND_HALF: 22,
  FIRST_HALF_OVERTIME: 23,
  AWAITING_OVERTIME: 24,
  HALFTIME_OVERTIME: 25,
  SECOND_HALF_OVERTIME: 26,
  FINAL_OVERTIME: 27,
  AWAITING_PENALTIES: 28,
  PENALTIES: 29,
  FINAL_PENALTIES: 30
};

export const AMERICAN_FOOTBALL_MATCH_STATUS = {
  ...MATCH_STATUS,
  FIRST_QUARTER: 40,
  END_FIRST_QUARTER: 41,
  SECOND_QUARTER: 42,
  HALFTIME: 43,
  THIRD_QUARTER: 44,
  END_THIRD_QUARTER: 45,
  FOURTH_QUARTER: 46,
  END_FOURTH_QUARTER: 47,
  OVERTIME: 48,
  FINAL_OVERTIME: 49
};

export const LABELS = {
  [MATCH_STATUS.NOT_STARTED]: '',
  [MATCH_STATUS.FINAL]: 'Fim',
  [MATCH_STATUS.DELAYED]: 'Adiado',
  [MATCH_STATUS.CANCELLED]: 'Canc',
  [MATCH_STATUS.OTHER]: 'Outro',
  [FOOTBALL_MATCH_STATUS.FIRST_HALF]: '1T',
  [FOOTBALL_MATCH_STATUS.HALFTIME]: 'Int',
  [FOOTBALL_MATCH_STATUS.SECOND_HALF]: '2T',
  [FOOTBALL_MATCH_STATUS.FIRST_HALF_OVERTIME]: '1TE',
  [FOOTBALL_MATCH_STATUS.HALFTIME_OVERTIME]: 'Int TE',
  [FOOTBALL_MATCH_STATUS.SECOND_HALF_OVERTIME]: '2TE',
  [FOOTBALL_MATCH_STATUS.FINAL_OVERTIME]: 'Fim TE',
  [FOOTBALL_MATCH_STATUS.AWAITING_PENALTIES]: 'Pen',
  [FOOTBALL_MATCH_STATUS.PENALTIES]: 'Pen',
  [FOOTBALL_MATCH_STATUS.FINAL_PENALTIES]: 'Fim Pen',
  [AMERICAN_FOOTBALL_MATCH_STATUS.FIRST_QUARTER]: '1Q',
  [AMERICAN_FOOTBALL_MATCH_STATUS.END_FIRST_QUARTER]: 'Fim 1Q',
  [AMERICAN_FOOTBALL_MATCH_STATUS.SECOND_QUARTER]: '2Q',
  [AMERICAN_FOOTBALL_MATCH_STATUS.HALFTIME]: 'HT',
  [AMERICAN_FOOTBALL_MATCH_STATUS.THIRD_QUARTER]: '3Q',
  [AMERICAN_FOOTBALL_MATCH_STATUS.END_THIRD_QUARTER]: 'Fim 3Q',
  [AMERICAN_FOOTBALL_MATCH_STATUS.FOURTH_QUARTER]: '4Q',
  [AMERICAN_FOOTBALL_MATCH_STATUS.END_FOURTH_QUARTER]: 'Fim 4!',
  [AMERICAN_FOOTBALL_MATCH_STATUS.OVERTIME]: 'OT',
  [AMERICAN_FOOTBALL_MATCH_STATUS.FINAL_OVERTIME]: 'Fim OT'
};

export const STOPPED_GAME = [
  MATCH_STATUS.NOT_STARTED,
  MATCH_STATUS.FINAL,
  MATCH_STATUS.DELAYED,
  MATCH_STATUS.CANCELLED,
  MATCH_STATUS.OTHER,
  FOOTBALL_MATCH_STATUS.HALFTIME,
  FOOTBALL_MATCH_STATUS.AWAITING_OVERTIME,
  FOOTBALL_MATCH_STATUS.HALFTIME_OVERTIME,
  FOOTBALL_MATCH_STATUS.FINAL_OVERTIME,
  FOOTBALL_MATCH_STATUS.AWAITING_PENALTIES,
  FOOTBALL_MATCH_STATUS.FINAL_PENALTIES,
  AMERICAN_FOOTBALL_MATCH_STATUS.END_FIRST_QUARTER,
  AMERICAN_FOOTBALL_MATCH_STATUS.HALFTIME,
  AMERICAN_FOOTBALL_MATCH_STATUS.END_THIRD_QUARTER,
  AMERICAN_FOOTBALL_MATCH_STATUS.END_FOURTH_QUARTER,
  AMERICAN_FOOTBALL_MATCH_STATUS.FINAL_OVERTIME
];
