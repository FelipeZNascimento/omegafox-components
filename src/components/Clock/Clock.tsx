import { DateTime } from 'luxon';
import classNames from 'classnames';

import { IClockProps } from './types';

import {
  LABELS,
  MATCH_STATUS,
  FOOTBALL_MATCH_STATUS,
  STOPPED_GAME
} from './constants';
import styles from './Clock.module.scss';
import matchLayer from '../../img/match_layer.png';
import { BET_VALUES } from 'components/Match/constants';

export const Clock = ({
  betValue,
  clock,
  isExpandable = false,
  isExpanded,
  isMatchEditable = false,
  timestamp
}: IClockProps) => {
  const formattedDate = DateTime.fromSeconds(timestamp)
    .setLocale('pt-Br')
    .toFormat("HH'h'mm");

  const errorFlagClass = classNames(styles.toggle, {});
  const toggleClass = classNames(styles.toggle, {
    [styles.toggleGreen]: betValue === BET_VALUES.FULL,
    [styles.toggleBlue]: betValue === BET_VALUES.HALF,
    [styles.toggleLightBlue]: betValue === BET_VALUES.MINIMUN,
    [styles.toggleRed]: betValue === BET_VALUES.MISS,
    [styles.toggleNeutral]: betValue === null
  });

  const renderCorner = () => {
    if (isMatchEditable) {
      return <div className={errorFlagClass} />;
    }

    if (isExpandable) {
      return <div className={toggleClass}>{isExpanded ? '-' : '+'}</div>;
    }

    return <div className={toggleClass} />;
  };

  const dateClass = classNames(styles.info, {
    [styles.infoDate]: clock.status === MATCH_STATUS.NOT_STARTED,
    [styles.infoClock]: clock.status !== MATCH_STATUS.NOT_STARTED
    // [styles.infoClockAnimated]: !STOPPED_GAME.includes(clock.status)
  });

  const clockClass = classNames({
    [styles.infoClockAnimated]: !STOPPED_GAME.includes(clock.status)
  });

  const renderDate = () => {
    if (clock.status === MATCH_STATUS.NOT_STARTED) {
      return formattedDate;
    } else {
      if (STOPPED_GAME.includes(clock.status)) {
        return LABELS[clock.status];
      } else {
        return (
          <>
            <span className={styles.infoClockStatus}>
              {clock.status !== FOOTBALL_MATCH_STATUS.PENALTIES &&
                LABELS[clock.status]}
            </span>
            <span className={clockClass}>
              {clock.status !== FOOTBALL_MATCH_STATUS.PENALTIES &&
                `${clock.time}'`}
              {clock.status === FOOTBALL_MATCH_STATUS.PENALTIES &&
                LABELS[clock.status]}
            </span>
          </>
        );
      }
    }
  };

  return (
    <div
      className={styles.dateContainer}
      style={{
        background: `url(${matchLayer}) #eceff1`
      }}
    >
      {renderCorner()}
      <div className={dateClass}>{renderDate()}</div>
    </div>
  );
};
