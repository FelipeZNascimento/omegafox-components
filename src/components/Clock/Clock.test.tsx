import { render } from '@testing-library/react';

import { Clock } from './Clock';
import { IClockProps } from './types';
import { footballClock, matchInfo } from '../mocks';

describe('ClockComponent', () => {
  const renderComponent = ({
    betStatus,
    clock,
    isExpandable,
    isExpanded,
    isMatchEditable,
    timestamp
  }: IClockProps) =>
    render(
      <Clock
        isExpandable={isExpandable}
        isMatchEditable={isMatchEditable}
        betStatus={betStatus}
        clock={clock}
        isExpanded={isExpanded}
        timestamp={timestamp}
      />
    );

  it('should render', () => {
    const { container } = renderComponent({
      isExpandable: true,
      isMatchEditable: false,
      betStatus: 'neutral',
      clock: footballClock,
      isExpanded: false,
      timestamp: matchInfo.timestamp
    });

    expect(container).toMatchSnapshot();
  });
});
