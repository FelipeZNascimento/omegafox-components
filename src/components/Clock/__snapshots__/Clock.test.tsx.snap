// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClockComponent should render 1`] = `
<div>
  <div
    class="dateContainer"
    style="background: rgb(236, 239, 241) url();"
  >
    <div
      class="toggle toggleGreen"
    >
      +
    </div>
    <div
      class="info infoClock"
    >
      <span
        class="infoClockStatus"
      >
        1T
      </span>
      <span
        class="infoClockAnimated"
      >
        45'
      </span>
    </div>
  </div>
</div>
`;
